src/core/crypto/singleKey.ts

```diff
@@ -0,0 +1,194 @@
+import { Deserializer, Serializer } from "../../bcs";
+import {
+  AnyPublicKeyVariant,
+  AnySignatureVariant,
+  SigningScheme as AuthenticationKeyScheme,
+  SigningSchemeInput,
+} from "../../types";
+import { AuthenticationKey } from "../authenticationKey";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import { AccountPublicKey, CryptoPublicKey, VerifyAccountSignatureArgs } from "./publicKey";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+import { AccountSignature, CryptoSignature } from "./signature";
+
+/**
+ * Represents any public key supported by Aptos.
+ *
+ * Since [AIP-55](https://github.com/aptos-foundation/AIPs/pull/263) Aptos supports
+ * `Legacy` and `Unified` authentication keys.
+ *
+ * Any unified authentication key is represented in the SDK as `AnyPublicKey`.
+ */
+export class AnyPublicKey extends AccountPublicKey {
+  /**
+   * Reference to the inner public key
+   */
+  public readonly publicKey: CryptoPublicKey;
+
+  /**
+   * Index of the underlying enum variant
+   */
+  public readonly variant: AnyPublicKeyVariant;
+
+  // region Constructors
+
+  constructor(publicKey: CryptoPublicKey) {
+    super();
+    this.publicKey = publicKey;
+    if (publicKey.signatureScheme === SigningSchemeInput.Ed25519) {
+      this.variant = AnyPublicKeyVariant.Ed25519;
+    } else if (publicKey.signatureScheme === SigningSchemeInput.Secp256k1Ecdsa) {
+      this.variant = AnyPublicKeyVariant.Secp256k1;
+    } else {
+      throw new Error("Unsupported public key type");
+    }
+  }
+
+  // endregion
+
+  // region AccountPublicKey
+
+  /**
+   * Authentication scheme
+   */
+  public readonly authenticationScheme = AuthenticationKeyScheme.SingleKey;
```

Comment:
can remove to the top of the class

Reply:
You mean move?
`AnyPublicKey` extends `AccountPublicKey` and it is required to implement some methods and getters.
I put all such members within the `AccountPublicKey` region so it's nicely organized.
Similarly there's a `Serializable` region

Reply:
sorry, I meant move to the top of the class with all other class variables definitions.

I understand the benefits of having those region comments, but I feel we implement it in some specific places and end up with the SDK not being consistent. I would argue we should have/implement the region comment helper in a new and fresh PR where we implement it throughout the SDK and not only in some random files we happen to work on in a PR